SpatialGDK/Source/Private/EngineClasses/SpatialGameInstance.cpp

```diff
@@ -92,6 +102,32 @@ bool USpatialGameInstance::StartGameInstance_SpatialGDKClient(FString& Error)
 	return bOk;
 }
 
+FURL USpatialGameInstance::GetInitialGameURL() const
+{
+	// Load the default URL from config files.
+	FURL DefaultURL;
+	DefaultURL.LoadURLConfig(TEXT("DefaultPlayer"), GGameIni);
+
+	const TCHAR* Cmd = FCommandLine::Get();
+
+	// Load up the default game map settings.
+	const UGameMapsSettings* GameMapsSettings = GetDefault<UGameMapsSettings>();
+	const FString& DefaultMap = GameMapsSettings->GetGameDefaultMap();
+
+	// Check to see if there's a command line URL override.
+	FString URLToLoad;
+	if (!FParse::Token(Cmd, URLToLoad, 0) || **URLToLoad == '-')
+	{
+		// If not, use the default map and options.
+		URLToLoad = DefaultMap + GameMapsSettings->LocalMapOptions;
+	}
+
+	FURL URL(&DefaultURL, *URLToLoad, TRAVEL_Partial);
+	checkf(URL.Valid == 1, TEXT("Initial URL was invalid: %s"), *URL.ToString());
```

Comment:
If the user sets the `DefaultPlayer` URL to something that's invalid and there is no commandline override, does that mean that this piece of code will prevent the instance from launching?
Might be better to log an error and just switch to `URL = WorldContext->LastURL;` in that case?

Reply:
Good point. I borrowed this mostly from engine code, and that code just considers it a browse failure for invalid URLs. Will change to do the same.